devito/types/tensor.py

```diff
@@ -62,15 +104,19 @@ def __subfunc_setup__(cls, *args, **kwargs):
         stagg = kwargs.get("staggered", None)
         name = kwargs.get("name")
         symm = kwargs.get('symmetric', True)
+        diag = kwargs.get('diagonal', False)
+
+        funcs = []
```

Comment:
actually, this line can be dropped I think?

and below in the for loop you may use `funcs` instead of `funcs2` as variable name

Reply:
reminder

Reply:
Lemme recheck it, think was tricky to only use one.